h/models.py

```diff
@@ -115,13 +115,25 @@ def __acl__(self):
     }
     __analysis__ = {
         'filter': {
-            'uri': {
+            'uri_index': {
                 'type': 'pattern_capture',
                 'preserve_original': '1',
                 'patterns': [
                     '([^\\/\\?\\#\\.]+)',
                     '([a-zA-Z0-9]+)(?:\\.([a-zA-Z0-9]+))*',
                     '([a-zA-Z0-9-]+)(?:\\.([a-zA-Z0-9-]+))*',
+                    '://(.+)',
+                    '://(.+)\\?',
+                    '://(.+)\\#'
+                ]
+            },
+            'uri_search': {
+                'type': 'pattern_capture',
+                'preserve_original': '1',
+                'patterns': [
+                    '://(.+)',
+                    '://(.+)\\?',
```

Comment:
Does this also remove the query string?
Ie. does this make http://site.com/content?id=1 equivalent with  http://site.com/content?id=2 ?
If yes, do we want that?
I don't think so.

Removing the hash part is warranted; removing the query string is much more controversial. 
I would guess that this would give us way more false positives, compared to the cases when it's actually useful.